components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -73,20 +80,41 @@ const hasDecimalsOverflow = (
   return decimalPlaces !== null && decimalPlaces > 0
 }
 
+const getTokenFromParam = (
+  contractOrSymbol: string,
+  network: BraveWallet.NetworkInfo,
+  tokenList: BraveWallet.BlockchainToken[]
+) => {
+  return tokenList.find(
+    (token) =>
+      (token.chainId === network.chainId &&
+        token.coin === network.coin &&
+        token.contractAddress.toLowerCase() ===
+          contractOrSymbol.toLowerCase()) ||
+      (token.chainId === network.chainId &&
+        token.coin === network.coin &&
+        token.contractAddress === '' &&
+        token.symbol.toLowerCase() === contractOrSymbol.toLowerCase())
+  )
+}
+
 export const useSwap = () => {
-  // State
-  const [fromAccount, setFromAccount] = useState<
-    BraveWallet.AccountInfo | undefined
-  >(undefined)
-  const [fromToken, setFromToken] = useState<
-    BraveWallet.BlockchainToken | undefined
-  >(undefined)
-  const [toAccountId, setToAccountId] = useState<
-    BraveWallet.AccountId | undefined
-  >(undefined)
-  const [toToken, setToToken] = useState<
-    BraveWallet.BlockchainToken | undefined
-  >(undefined)
+  // routing
+  const query = useQuery()
+  const history = useHistory()
+
+  // Queries
+  // FIXME(onyb): what happens when defaultFiatCurrency is empty
+  const { data: defaultFiatCurrency } = useGetDefaultFiatCurrencyQuery()
+  const { data: supportedNetworks } = useGetSwapSupportedNetworksQuery()
+  const { data: fullTokenList } = useGetCombinedTokensListQuery()
+  const { account: fromAccount } = useAccountFromAddressQuery(
+    query.get('fromAccountId') ?? undefined
+  )
+  const { account: toAccount } = useAccountFromAddressQuery(
```

Comment:
Prefer not adding the assumption that `toAccountId` will always be a wallet address. We should have a line like this:

```ts
const toAccountId = makeAccountIdFromUniqueKey(
  query.get('toAccountId')
)
```

Reply:
Actually, if you want to take it even further, the right thing to do here is have these two query params representing the To account/address:
- `toCoin` (fromCoin is not necessary)
- `toAddress`

Reply:
Replaced the `toAccountId` `param` with `toCoin` and `toAddress`.

For now we will have a `toAccountId` variable declared as `fromAccountId` values.